src/app/ReadClient.cpp

```diff
@@ -504,7 +504,12 @@ CHIP_ERROR ReadClient::ProcessReportData(System::PacketBufferHandle && aPayload)
     err = report.GetSubscriptionId(&subscriptionId);
     if (CHIP_NO_ERROR == err)
     {
-        if (mIsPrimingReports)
+        if (!IsSubscriptionType())
+        {
+            err = CHIP_ERROR_INVALID_ARGUMENT;
+            goto exit;
+        }
```

Comment:
Perhaps:
```suggestion
        VerifyOrExit(IsSubscriptionType(), err = CHIP_ERROR_INVALID_ARGUMENT);
```
but either way.